wiki/Tutorials/ JupyterHub/JupyterHub.md

```diff
@@ -0,0 +1,190 @@
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: data
+spec:
+  capacity:
+    storage: 1000Gi
+  accessModes:
+    - ReadWriteMany
+  nfs:
+    server: #TODO: set url to NFS server (e.g. manila-dmz-prod.isi.denbi.bihealth.org)
+    path: #TODO: set path to NFS share
+  mountOptions:
+    - nfsvers=3
+```
+
+* The next command creates the share in Kubernetes:
+
+```bash
+kubectl apply --namespace jhub -f data-pv.yaml
+```
+
+* To mount the share into JupyterHub, we also need a persistent volume claim. The content below needs to be stored in the file `data-pvc.yaml`
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: data-pvc
+spec:
+  accessModes:
+    - ReadWriteMany
+  storageClassName: ""
+  resources:
+    requests:
+      storage: 1000Gi
+```
+
+* Finally, the next command creates the volume claim in Kubernetes:
+
+```bash
+kubectl apply --namespace jhub -f data-pvc.yaml
+```
+
+### Network
+
+In the Berlin node, a floating-ip with public access needs specifically be assigned to your project. Please ask the Berlin cloud admin team for the provision of a public floating ip via mail: denbi-cloud@bih-charite.de . We will use the floating IP `xxx.xxx.xxx.xxx` in the example and the respective public IP is `xxx.xxx.xxx.xxx`. Both adresse are just used for the example and should not really exist.
+
+### DNS & encryption
+
+To access the portal from the internet via https, we also need a DNS entry and a TLS certificate. In some cases, a DNS entry for the bihealth.org domain can be requested for a de.NBI project at the Berlin site. Please contact the site cloud admins to get information for your specific project. In the example here, we use the URL `my_platform.bihealth.org`. Next, we also need a certificate for encrypted communication. For example, a certficate can be created by letsencrypt (https://letsencrypt.org/de/getting-started/). The keyfile and the certificate file need to be stored on the `deploy-node`.  Both files are stored as a secret in Kubernetes so that we can use them in the JupyterHub deployment. Name of the secret is `jupyter-tls`.
+
+```console
+kubectl create secret --namespace=jhub tls jupyter-tls --key=$PATH_TO_KEY --cert=$PATH_TO_CERT
+```
+
+### OIDC
+
+We recommend using GitHub for user authentication, because it is easy to configure and also easy to manage users. Any other OIDC provider such as LifeScience AAI could be used here as well. See here (TODO) for an full tutorial on how to create an oauth2 appp for GitHub (TODO). In summary, first we need to create an organization, then we can create an OAuth app and include the url of our JupyterHub platform and add the path to the callbak site, which will be e.g. `my_platform.bihealth.org/hub/oauth_callback`. After that, adding users to the GitHub organization will be sufficient to autorize them for using the JupyterHub platform.
```

Comment:
```suggestion
We recommend using GitHub for user authentication, because it is easy to configure and also easy to manage users. Any other OIDC provider such as LifeScience AAI could be used here as well. See here (TODO) for an full tutorial on how to create an oauth2 app for GitHub (TODO). In summary, first we need to create an organization, then we can create an OAuth app and include the url of our JupyterHub platform and add the path to the callback site, which will be e.g. `my_platform.bihealth.org/hub/oauth_callback`. After that, adding users to the GitHub organization will be sufficient to authorize them for using the JupyterHub platform.
```

> See here (TODO) for an full tutorial on how to create an oauth2 appp for GitHub (TODO)

References are missing.